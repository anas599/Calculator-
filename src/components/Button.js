// eslint-disable-next-line import/no-extraneous-dependencies
import PropTypes from 'prop-types';

function Button({ onClick, keyPad, className }) {
  return (
    <button type="button" onClick={onClick} className={className}>
      {keyPad}
    </button>
  );
}

Button.propTypes = {
  onClick: PropTypes.func.isRequired,
  keyPad: PropTypes.string.isRequired,
  className: PropTypes.string.isRequired,
};

export default Button;
