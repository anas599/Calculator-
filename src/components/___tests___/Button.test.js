import renderer from 'react-test-renderer';
import Button from '../Button';

describe('snapshot testing', () => {
  it('first snapshot', () => {
    const keyClick = () => {};
    const itemRender = renderer
      .create(
        <Button keyPad="AC" onClick={keyClick} className="greyKey" />,
      )
      .toJSON();
    expect(itemRender).toMatchSnapshot();
  });
});
