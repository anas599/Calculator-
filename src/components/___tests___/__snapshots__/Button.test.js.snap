// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing first snapshot 1`] = `
<button
  type="button"
/>
`;
